### DISCUSSION TITLE
Custom document IDs in RavenDB ETL

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#18104] [Q&A] [answered]
I understand there is no way to customize the IDs for ETL-ed documents. When the source and target collection names match, the IDs are preserved. Otherwise, they are constructed as described here: https://ravendb.net/docs/article-page/6.0/csharp/server/ongoing-tasks/etl/raven#documents-identifiers .

For our use case, it would be very beneficial to be able to customize the ID format.

We combine data from multiple databases into one, with some transformations, and would like to be able to load the documents based on their original ID in the target database. This is currently not easy, as it's impossible to determine the ID of the document in the target database based on the original ID (we ETL to a different collection for each source database, as there might be ID conflicts).

We can avoid conflicts by using a prefix for each source database, so let's say company-12345 would become countryA-12345 and countryB-12345, depending on the source database.

Is it possible to achieve this using RavenDB ETL? Do you think this scenario makes sense? Can it be achieved in an alternative way?

**Comment by arekpalinski** [comment, marked as answer]
Hi @krivulcik,

Are you aware of the ability to define the document ID postfix?

![image](https://github.com/ravendb/ravendb/assets/376551/499fcd14-34d0-443e-a285-d611525cda2f)

https://issues.hibernatingrhinos.com/issue/RavenDB-13953/Allow-to-define-document-id-postfix-in-RavenDB-ETL

It was added exactly for the scenario where multiple sources (databases) will ETL to the same database. This way you'll be able to keep the same same collection, hence IDs will stay as in original database + the postfix that you define.

**Reply by krivulcik** [reply]
Thanks, @arekpalinski, that's exactly what we need. We didn't notice that before.
